Update the channels guide to match Phoenix 1.0.2

jeffkreeftmeijer opened the pull request:
Since some things have changed since the channels guide was written, I updated some of the code samples and a bit of the description so the guide matches Phoenix 1.0.2.

To make sure the code in the guide matches the current Phoenix version, I started out by generating a Phoenix 1.0.2 app named `HelloPhoenix` (like in the getting started guide) and worked from there. This means some small things changed, like the added `, otp_app: :hello_phoenix` in `lib/hello_phoenix/endpoint.ex

I tried to use as much of the generated code (like `lib/hello_phoenix/endpoint.ex`, `web/socket/user_socket.ex`, and `web/static/js/socket.js`) as possible. This means, for example, that all socket-related javascript code moved from `web/static/js/app.js` to `web/static/js/socket.js`, simply because a lot of the guide's code whas already generated there.
 
Although this is quite a big change, I tried to keep the changes as small as possible. I have fixed a typo or two along the way, though.

Oh, and here's a demo app of the guide's exact end result: https://github.com/jeffkreeftmeijer/phoenix_channels_example. It'll probably be useful to have this app around to help us update the guide when future versions of Phoenix are released.

chrismccord commented on G_channels.md:
```diff
@@ -92,10 +89,10 @@ The first priority of your channels is to authorize clients to join a given topi
+  def join("rooms:lobby", _message, socket) do
     {:ok, socket}
   end
-  def join("rooms:" <> _private_room_id, _auth_msg, socket) do
```
the subtopic is the private room id, so lets' make this line `def join("rooms:" <> _private_room_id, _params, socket) do`

chrismccord commented on G_channels.md:
````diff
@@ -103,22 +100,31 @@ end
 ```javascript
-import {Socket} from "deps/phoenix/web/static/js/phoenix"
+...
+socket.connect({token: window.userToken})
````
passing params to connect is deprecated. They should now be passed as a `params:` argument to the Socket constructor

chrismccord commented on G_channels.md:
```diff
@@ -195,10 +200,10 @@ We handle incoming events with `handle_in/3`. We can pattern match on the event
     {:ok, socket}
   end
-  def join("rooms:" <> _private_room_id, _auth_msg, socket) do
+  def join("rooms:" <> _private_subtopic, _message, _socket) do
```
Same here `def join("rooms:" <> _private_room_id, _params, socket) do`

jeffkreeftmeijer commented:
Hey @chrismccord, thanks for your input. I've updated the `socket.connect()` call and changed `join/3` to `join("rooms:" <> _private_subtopic, _message, _socket)`. I've ignored the `_socket` argument to prevent getting an unused variable warning;

    web/channels/room_channel.ex:7: warning: variable socket is unused

lancehalvorsen commented:
Hi @jeffkreeftmeijer, sorry I'm a little late here. Yesterday was a travel day for me. I really appreciate the way you've matched the tone and usage in the rest of the guides. <3 <3 <3

Thanks so much for this contribution!
